db/db_impl/db_impl_secondary.cc

```diff
@@ -1008,6 +1008,10 @@ Status DBImplSecondary::ParseCompactionProgressFile(
     return s;
   }
 
+  if (!reader_status.ok()) {
+    return reader_status;
```

Comment:
To please ASSERT_STATUS_CHECK. Missing it (i.e, not entering the loop of compaction_progress_reader.ReadRecord(&slice, &record)) is not a bug because we have the blanket statement below `if (progress_builder.HasAccumulatedSubcompactionProgress())`